### DISCUSSION TITLE
How to get messages from group chat by ChannelID

### DISCUSSION THREAD

**Discussion by nikoole** [xelaj/mtproto#85] [Q&A]
Hi...
I just confused. How can i receive last messages from Channel by ChannelID ?

**Comment by quenbyako** [comment]
Hey @nikoole!

Sorry for long response, actualty you can fetch any data from chats, channels, soupergroups etc. with sending retrieving method with `inputSomething` object (like [this one](https://core.telegram.org/method/channels.getMessages)): inputs are REALLY tricky part of telegram api, but, well, when you to once understand it your life become more understandable (and you will hate Durov brothers all life, lol)

To init `inputChannel` you must get list of your current channels from [this method, f.e.](https://core.telegram.org/method/messages.getAllChats) and save list of chats with data somewhere (in map maybe, or on fs, doesn't matter)

Next, you must get channel id **and channel hash** of channel what you need (this is sorta security feature which is hard and bad as fuck) and unite them into `InputChannelObj` struct. After that, you can request any data with this object: take messages, edit channel photo, set rules, kick participants, everything.

Yeeeeah, i know, this sounds really weird, but i think everyone must be argue to telegram developers with that.
